### DISCUSSION TITLE
Metrics of services with the same name compounded in the Inbound Metrics tab

### DISCUSSION THREAD

**Discussion by spockz** [kiali/kiali#3646] [Q&A] [answered]
In our testing setup we have a single server (`s1-service-a)` and three consumers. two consumers in the same namespace, `s1-client-x` and `s1-client-y`. There is another namespace that also contains a consumer, also called `s1-client-y`. All three consumers do 1RPS continuously. The Graph view correctly distinguishes between the different consuming workloads and adds the namespace identifier. However, in the "Inbound metrics" tab of `s1-service-a`, I only see two consumers, and the load from services called `s-client-y` are added to each other. At the least this is inconsistent. Personally, I consider this a bug and the inbound metrics should distinguish using the namespace as well.

I've added screenshots of the situation as well.
<img width="1710" alt="compounded-metrics-graph-view" src="https://user-images.githubusercontent.com/54579/106192859-066d6080-61ad-11eb-9110-8efe5f97bfb4.png">

<img width="1710" alt="kiali-workloads-view" src="https://user-images.githubusercontent.com/54579/106192868-08cfba80-61ad-11eb-8242-2e6a1673b76d.png">
<img width="1710" alt="inbound-metrics-view" src="https://user-images.githubusercontent.com/54579/106192873-0a00e780-61ad-11eb-95f6-7bf885bbf557.png">

**Comment by jshaughn** [comment, marked as answer]
@jotak Not sure if this is expected behavior, or not.  What do you think?   Looking at [this code](https://github.com/kiali/kiali/blob/master/models/dashboards.go#L122-L125) I'm not sure things are currently set up to discriminate on namespace.

**Reply by jotak** [reply]
Indeed, that looks like a bug, let me investigate

**Reply by jotak** [reply]
I miss a "convert to bug" button in github's discussion :-)
EDIT actually I've found it

**Reply by jotak** [reply]
Issue reference: https://github.com/kiali/kiali/issues/3650

**Comment by spockz** [comment]
Fixed by #3650.
